### DISCUSSION TITLE
unsubscribe from Observable

### DISCUSSION THREAD

**Discussion by theSilverFisch** [spierala/mini-rx-store#108] [Q&A] [answered]
Hi guys,

I wonder how to unsubscribe from an observable as I don't seem to find the .unsubscribe() function. I'm using the mini-rx-store together with vue3.
What I do is the following:

`const $store = inject('$store')`
`const state$ = $store.select(state => state.currentPage)`
`state$.subscribe(val => {`
`state.value = val`
`})`

so state$ is the observable which I get via select and it works as it should. But in my unmount function I'd now like to unsubscribe from that. Any advice?

Regards 

Christian

**Comment by theSilverFisch** [comment, marked as answer]
ok I just answered it myself. I need to assign 
`const subscriber = state$.subscribe(....)`

then I can call 
`subscriber.unsubscribe()`

**Comment by spierala** [comment]
Hi @theSilverFisch, cool that you are using MiniRx with Vue!

Yes, you found the right answer.
In Vue you could simplify your code by using https://vueuse.org/rxjs/usesubscription/ or https://vueuse.org/rxjs/useobservable/ (both come from this lib: https://vueuse.org/rxjs/README.html)

Maybe, also vue-rx could help to manage your subscriptions: https://github.com/vuejs/vue-rx
But the last publish of the vue-rx lib is already 3 years old. Not sure if it is still actively developed.
